model/src/main/java/com/javahelp/model/user/User.java

```diff
@@ -0,0 +1,89 @@
+package com.javahelp.model.user;
+
+/**
+ * A class for a user entity
+ */
+public class User {
+
+    /**
+     * The ID of the user account (randomly generated and immutable)
```

Comment:
I think we should change the javadoc here as well as to keep it clear that it's immutable ONLY when it's assigned and saved to the database